### DISCUSSION TITLE
[RFC] Spin support `percent`

### DISCUSSION THREAD

**Discussion by zombieJ** [ant-design/ant-design#48647] [RFCs - archive]
# 📋 RFC

## Summary

Spin component support `percent` prop as other loading related API for progress showing.

![Kapture 2024-04-26 at 16 16 43](https://github.com/ant-design/ant-design/assets/5378891/cbd745a8-f392-41a8-a7a4-348c14fe0782)

![Kapture 2024-04-26 at 16 18 31](https://github.com/ant-design/ant-design/assets/5378891/4df105c5-3b06-40eb-9995-ff0b0644f4a3)

## Basic example

```tsx
<Spin percent={33}  />

<Spin percent={33} fullscreen />

<Spin percent={33} size="large" />
```

## Motivation

related RFC #48645 

Spin could provide more process info about loading status.

## Detailed

When `spinning`, it will take `percent` number for render process.
Otherwise will do nothing with this prop.

## Drawbacks

N/A

## Adoption strategy

feature in minor

**Comment by MadCcc** [comment]
和 tip 怎么兼容？

**Reply by zombieJ** [reply]
文案部分应该不影响这个，需要加个 tip renderProps ？

**Comment by afc163** [comment]
这不就是小尺寸的 progress？

https://ant.design/components/progress-cn#components-progress-demo-size

**Reply by zombieJ** [reply]
直接融进去？

**Reply by afc163** [reply]
感觉和 spin 没关系。

**Reply by zombieJ** [reply]
参考一下 gif，这个做替换对开发者而言是比较麻烦的。也无法做主题定制，比如 快乐工作 主题实现替换，就需要额外提供一个类似 Spin 功能的组件。

**Comment by afc163** [comment]
感觉需要加个自增的 precent 计算逻辑，不然业务上要自己算 percent 来做效果。

**Reply by zombieJ** [reply]
类似于这种？会一直走，但是永远不会走到头：

```tsx
<Spin
  percent="guess"
/>
```

**Reply by afc163** [reply]
对的，不然业务不太好写。

如果支持自动增长可以这样写：

```
percent={isDone ? 100 : 'auto'}
```

完成的时候从当前点一个 transition 动画走到 100%。

**Comment by afc163** [comment]
Button 也需要类似的能力，用于有进度的 loading 状态展示。

```
<Button loading={{ percent: number }} />
```

**Reply by zombieJ** [reply]
嗯，这个会出个新的 RFC
